src/app/charity/charity.component.ts

```diff
@@ -13,17 +15,37 @@ export class CharityComponent implements OnInit {
   campaigns: CampaignSummary[];
 
   constructor(
+    private datePipe: DatePipe,
     private pageMeta: PageMetaService,
     private route: ActivatedRoute,
   ) {}
 
   ngOnInit(): void {
-    this.campaigns = this.route.snapshot.data.campaigns;
+    this.campaigns = this.route.snapshot.data.campaigns.sort((a: CampaignSummary, b: CampaignSummary) => {
+      return new Date(b.endDate).getTime() - new Date(a.endDate).getTime();
+    });
 
     this.pageMeta.setCommon(
       this.campaigns[0] ? `${this.campaigns[0].charity.name} Campaigns` : 'Campaigns Archive',
       'Archive of Big Give match funded campaigns',
       null,
     );
   }
+
+  getPercentageRaised(childCampaign: CampaignSummary) {
+    CampaignService.percentRaisedOfIndividualCampaign(childCampaign);
+  }
+
+  getRelevantDateAsStr(campaign: CampaignSummary) {
```

Comment:
Definitely not a blocker but I think this is just complicated enough that it's worth making the type explicit instead of relying on inference.

```suggestion
  getRelevantDateAsStr(campaign: CampaignSummary): string | null {
```

Reply:
Seems reasonable – this helper was copied from the metacampaign page too, but I'm hesitant to abstract out one liners when there are only 2-3 instances